packages/clarity-js/src/layout/mutation.ts

```diff
@@ -145,6 +150,42 @@ async function process(): Promise<void> {
     task.stop(timer);
 }
 
+function track(m: MutationRecord, timer: Metric): string {
+  let value = m.target ? dom.get(m.target.parentNode) : null;
+  if (value) {
+    let inactive = time() > activePeriod;
+    // We use selector, instead of id, to determine the key because in some cases
+    // repeated mutations can cause elements to be destroyed and then recreated as new DOM nodes
+    // In those cases, IDs will change however the selector (which is relative to DOM xPath) remains the same
+    let key = [value.selector, m.attributeName, m.addedNodes ? m.addedNodes.length : 0, m.removedNodes ? m.removedNodes.length : 0].join();
+    // Initialize an entry if it doesn't already exist
+    history[key] = key in history ? history[key] : [0];
+    // Lookup any pending nodes queued up for removal, and process them now if we had suspended a mutation before
+    if (inactive === false && history[key][0] >= Setting.MutationSuspendThreshold) { processNodeList(history[key][1], Source.ChildListRemove, timer); }
+    // Update the counter
+    history[key][0] = inactive ? history[key][0] + 1 : 1;
+    // Return updated mutation type based on if we have already hit the threshold or not
+    if (history[key][0] === Setting.MutationSuspendThreshold) {
+      // Store a reference to removedNodes so we can process them later
```

Comment:
should we check if the node is inactive here? it looks like we are not processing the current changes to the element during the active period.

Reply:
We already do so within processNode function:
// Do not track this change if we are attempting to remove a node before discovering it
    if (source === Source.ChildListRemove && dom.has(node) === false) { return child; }

If the node is destroyed before we are able to process "remove" on it - we do handle it gracefully. 

---
In reply to: [643633409](https://github.com/microsoft/clarity/pull/137#discussion_r643633409) [](ancestors = 643633409)